include/RAJA/index/RangeSegment.hpp

```diff
@@ -87,8 +91,10 @@ struct TypedRangeSegment {
    * \param[in] begin the starting value (inclusive) for the range
    * \param[in] end the ending value (exclusive) for the range
    */
-  RAJA_HOST_DEVICE constexpr TypedRangeSegment(DiffT begin, DiffT end)
-      : m_begin(iterator{begin}), m_end(iterator{end})
+  using StripStorageT = strip_index_type_t<StorageT>;
+  RAJA_HOST_DEVICE constexpr TypedRangeSegment(StripStorageT begin, StripStorageT end)
+      : m_begin( begin > end ? throw std::runtime_error("RangeSegment : Ensure that begin <= end.") : iterator(begin)), 
```

Comment:
@rhornung67 I tried using RAJA_ABORT_OR_THROW previously, however when that is used instead of throwing an exception an error is thrown at compile time as it's expecting an object of type iterator not void.